sympy/geometry/point3d.py

```diff
@@ -513,10 +513,13 @@ def transform(self, matrix):
         geometry.entity.scale
         geometry.entity.translate
         """
-        from sympy.matrices.expressions import Transpose
-        x, y, z = self.args
-        m = Transpose(matrix)
-        return Point3D(*(Matrix(1, 4, [x, y, z, 1])*m).tolist()[0][:3])
+        if isinstance(matrix, Matrix) and matrix.shape == (4, 4):
```

Comment:
Same remark applies here.
Also, the check "is this a square matrix?" looks like it's generally useful. Maybe add an `is_square_matrix(size)` function.
I'm not sure that type checking in this way is a good idea. Python's approach is "duck typing", i.e. you simply call the appropriate functions and this will already do the crashing for you. So the `isinstance` check can go, it's enough to ask for the `shape` and let it crash if it's not a matrix (nor something that has a shape).
If an `isinstance` check is really desired, you should check for `MatrixBase`, because this function is supposed to work for `ImmutableMatrix` as well as for `Matrix` (I think).